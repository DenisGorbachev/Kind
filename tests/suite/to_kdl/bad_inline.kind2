NormalFn (a: U60) : U60
NormalFn a = (+ a a)

#inline
InlineFn (a: U60) : U60
InlineFn 0 = (- 0 1)
InlineFn 1 = (+ 2 2)
InlineFn n = n

BadInline : U60 {
  let bad_arg = 1
  let a = InlineFn 1
  let b = InlineFn ((x => x) 1)
  let c = InlineFn (NormalFn 0)
  let d = InlineFn (- 2 1)
  (+ a (+ b (+ c d)))
}