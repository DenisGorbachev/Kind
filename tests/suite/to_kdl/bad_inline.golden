Unable to match term Fun { name: Ident("InlineFn"), args: [App { func: Lam { name: Ident("x"), body: Var { name: Ident("x") } }, argm: Num { numb: 1 } }] } to any of the function's rules